src/internal/m365/exchange/contacts_restore.go

```diff
@@ -94,22 +99,40 @@ func restoreContact(
 	}
 
 	ctx = clues.Add(ctx, "item_id", ptr.Val(contact.GetId()))
-	collisionKey := api.ContactCollisionKey(contact)
 
-	if _, ok := collisionKeyToItemID[collisionKey]; ok {
+	var (
+		collisionKey = api.ContactCollisionKey(contact)
+		collisionID  string
+		replace      bool
```

Comment:
`delete` might be a better name as this is used to decide if we should delete the item. Same in other places.